opencl-aot/source/main.cpp

```diff
@@ -303,15 +338,100 @@ int main(int Argc, char *Argv[]) {
                               "OpenCL ahead-of-time (AOT) compilation tool");
 
   // step 1: perform checks for command line options
-  bool IsFileExists = false;
-  sys::fs::is_regular_file(OptInputBinary, IsFileExists);
-  if (!IsFileExists) {
-    std::cerr << "File " << OptInputBinary << " does not exist!" << '\n';
-    return OPENCL_AOT_FILE_NOT_EXIST;
+
+  // Setup default name of output file.
+  std::string OutputFileName((OptDevice == fpga_fast_emu) ? "aot-output.aocx"
+                                                          : "aot-output.bin");
+  if (OptOutputElf.getNumOccurrences())
+    OutputFileName = OptOutputElf;
+
+  std::vector<std::string> InputFileNames;
+  if (OptCommand.getNumOccurrences()) {
+    switch (OptCommand) {
+    case Commands::build:
+      // Ensure we get either -input or -binary
+      if (!OptBinaries.getNumOccurrences() && !OptInput.getNumOccurrences()) {
+        std::cerr << "-cmd=build needs -input or -binary option\n";
+        return OPENCL_AOT_OPTIONS_DEFICIENCY_FAILURE;
+      }
+      // Ensure we don't have both
+      if (OptBinaries.getNumOccurrences() && OptInput.getNumOccurrences()) {
+        std::cerr << "-cmd=link accepts only one of -input or -binary option\n";
+        return OPENCL_AOT_OPTIONS_COEXISTENCE_FAILURE;
+      }
+      // Get input file for -cmd=build
+      InputFileNames.push_back(OptInput.getNumOccurrences() ? OptInput
+                                                            : OptBinaries);
+      break;
+    case Commands::compile:
+      if (!OptInput.getNumOccurrences()) {
+        std::cerr << "-cmd=compile needs -input option\n";
+        return OPENCL_AOT_OPTIONS_DEFICIENCY_FAILURE;
+      }
+      // Get input file for -cmd=compile
+      InputFileNames.push_back(OptInput);
+      break;
+    case Commands::link:
+      // Ensure we get either -spv or -binary
+      if (!OptBinaries.getNumOccurrences() && !OptSPIRV.getNumOccurrences()) {
+        std::cerr << "-cmd=link needs -binary or -spv option\n";
+        return OPENCL_AOT_OPTIONS_DEFICIENCY_FAILURE;
+      }
+      // Ensure we don't have both
+      if (OptBinaries.getNumOccurrences() && OptSPIRV.getNumOccurrences()) {
+        std::cerr << "-cmd=link accepts only one of -binary or -spv option\n";
+        return OPENCL_AOT_OPTIONS_COEXISTENCE_FAILURE;
+      }
+
+      // Get input files for -cmd=link
+      if (OptSPIRV.getNumOccurrences()) {
+        InputFileNames.push_back(OptSPIRV);
+      } else {
+        // Split Binaries at ','s
+        std::stringstream ss(OptBinaries);
+        std::string item;
+        while (std::getline(ss, item, ',')) {
+          InputFileNames.push_back(item);
+        }
+      }
+
+      break;
+    }
+
+    // Output file name is specified by -ir if --device=fpga_fast_emu.
+    if (!OptIR.getNumOccurrences()) {
+      std::cerr << "-cmd=<value> needs -ir option\n";
+      return OPENCL_AOT_OPTIONS_DEFICIENCY_FAILURE;
+    }
```

Comment:
If `-ir` is only introduced for compatibility reasons, I suggest do not introduce this restriction and allow using regular `-o` even for `fpga_fast_emu` device and `-cmd` option, because such restriction basically blocks any transition to a regular `-o`

Reply:
Done.